task6/script.js

```diff
@@ -0,0 +1,412 @@
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '19',
+        description: 'test19',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Какой-то пользователь',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+    {
+        id: '20',
+        description: 'test20',
+        createdAt: new Date('2020-03-17T23:00:00'),
+        author: 'Какой-то пользователь',
+        photoLink: 'https://m2bob-forum.net/wcf/images/avatars/3e/2720-3e546be0b0701e0cb670fa2f4fcb053d4f7e1ba5.jpg',
+        hashTags: [
+            'hash','tag'
+        ],
+        likes: [
+            'Другой пользователь'
+        ]
+    },
+]
+
+let api = (function() {
+    let postSchema = {
+        id: val => typeof val === 'string' && val.length > 0,
+        description: val => typeof val === 'string' && val.length < 200,
+        createdAt: val => Object.prototype.toString.call(val) === '[object Date]',
+        author: val => typeof val === 'string' && val.length > 0,
+        photoLink: val => typeof val === 'string',
+        hashTags: val => Array.isArray(val),
+        likes: val => Array.isArray(val)
+    };
+
+    function validatePost(post) {
+        if((Math.abs(Object.keys(post).length - Object.keys(postSchema).length) == 1 && post['photoLink'] != null)
+            || Math.abs(Object.keys(post).length - Object.keys(postSchema).length) > 1) {
+            console.log('illegal arguments number');
+            return false;
+        }
+        return validateForm(post);
+    };
+
+    function validateForm(form) {
+        let errors = Object.keys(form)
+            .filter(key => !postSchema[key]?.(form[key]))
+            .map(key => new Error(key + ' is invalid!'));
+        if(errors.length > 0) {
+            errors.forEach(error => console.log(error.message))
+            return false;
+        }
+        return true;
+    }
+
+    function addPost(post) {
+        if(validatePost(post) && posts.find(p => post.id === p.id) == null) {
+            posts.push(post);
+            return true;
+        }
+        return false;
+    }
+    
+    function removePost(id) {
+        let startLength = posts.length;
+        posts = posts.filter(post => post.id !== id);
+        return startLength != posts.length;
+    }
+    
+    function getPost(id) {
+        return posts.find(post => post.id == id);
+    }
+
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        if(filterConfig != null) {
+            return posts.filter(p => (filterConfig.author == null || p.author === filterConfig.author) &&
+                (filterConfig.fromDate == null || p.createdAt >= filterConfig.fromDate) &&
+                (filterConfig.toDate == null || p.createdAt <= filterConfig.toDate) &&
```

Comment:
Minor. One of condition for from to should have the strict condition like >= && < or > && <=